addons/website_profile/controllers/main.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import werkzeug
+
+from odoo import http, modules, tools
+from odoo.http import request
+from odoo.tools import limited_image_resize
+
+
+class WebsiteProfile(http.Controller):
+
+    def _check_avatar_access(self, user_id, **post):
+        try:
+            user = request.env['res.users'].sudo().browse(user_id).exists()
+        except:
+            return False
+        if user:
+            return user.website_published and user.karma > 0
+        return False
+
+    @http.route([
+        '/profile/avatar/<int:user_id>',
+    ], type='http', auth="public", website=True, sitemap=False)
+    def user_profile_get_avatar(self, user_id=0, width=0, height=0, crop=False, avoid_if_small=False, upper_limit=False, **post):
+        check_result = self._check_avatar_access(user_id, **post)
+        fname = post.get('field', 'image_medium')
+        if check_result:
+            status, headers, content = request.env['ir.http'].sudo().binary_content(
+                model='res.users', id=user_id, field=fname,
+                default_mimetype='image/png')
+        else:
+            # TDE FIXME
+            status, headers, content = request.env['ir.http'].sudo().binary_content(
+                model='res.users', id=user_id, field=fname,
+                default_mimetype='image/png')
```

Comment:
whut ? if and else are doing the same thing. In Else, remove the sudo

Reply:
Sudo removed. It was for debugging. Probably.